app/models/document.rb

```diff
@@ -46,6 +54,51 @@ def facet_options(facet)
     finder_schema.options_for(facet)
   end
 
+  def format_specific_metadata
```

Comment:
This is nitpicky, and I don't know if it solves the problem totally, but at the very least, for consistency with `humanized_attributes` it'd be good to use `inject`*.  I reckon the `.map { ... }.reduce({}, :merge)` pattern is fairly confusing.

*`.each.with_object({})` in both might be even clearer though.